### ISSUE TITLE
[SOLVED] iOS Build Crash after upgrade from 0.60.4 to 0.62.2

### ISSUE THREAD

**Issue by cbothra** [react-native-community/upgrade-support#63]
Hi Folks,

I recently upgraded my react-native project from 0.60.4 to 0.62.2. I added all the dependencies and resolved all the errors thrown in terminal one by one and I was successfully able to compile the project. Now when I run the build in XCode the app crashes with following log in XCode:
```
2020-04-16 16:25:50.635 [error][tid:com.facebook.react.JavaScript] TypeError: undefined is not an object (evaluating 'R.ART.Shape')
2020-04-16 16:25:50.641 [fatal][tid:com.facebook.react.ExceptionsManagerQueue] Unhandled JS Exception: TypeError: undefined is not an object (evaluating 'R.ART.Shape')
2020-04-16 16:25:50.642 [error][tid:com.facebook.react.JavaScript] Invariant Violation: Module AppRegistry is not a registered callable module (calling runApplication)
2020-04-16 16:25:50.686677+0530 MY_APP[14817:106507] *** Terminating app due to uncaught exception 'RCTFatalException: Unhandled JS Exception: TypeError: undefined is not an object (evaluating 'R.ART.Shape')', reason: 'Unhandled JS Exception: TypeError: undefined is not an object (evaluating 'R.ART.Shape'), stack:
<unknown>@875:258
v@2:1473
<unknown>@874:228
v@2:1473
<unknown>@873:242
v@2:1473
<unknown>@872:150
v@2:1473
<unknown>@870:228
v@2:1473
<unknown>@868:342
v@2:1473
<unknown>@394:471
v@2:1473
<unknown>@6:70
v@2:1473
d@2:875
global code@1220:3
'
*** First throw call stack:
(
	0   CoreFoundation                      0x000000010c4601bb __exceptionPreprocess + 331
	1   libobjc.A.dylib                     0x000000010a846735 objc_exception_throw + 48
	2   MY_APP                                0x0000000106f0426a RCTFormatError + 0
	3   MY_APP                                0x0000000106f876fc -[RCTExceptionsManager reportFatal:stack:exceptionId:suppressRedBox:] + 519
	4   MY_APP                                0x0000000106f88197 -[RCTExceptionsManager reportException:] + 1669
	5   CoreFoundation                      0x000000010c46703c __invoking___ + 140
	6   CoreFoundation                      0x000000010c4644d5 -[NSInvocation invoke] + 325
	7   CoreFoundation                      0x000000010c464926 -[NSInvocation invokeWithTarget:] + 54
	8   MY_APP                                0x0000000106f36c51 -[RCTModuleMethod invokeWithBridge:module:arguments:] + 612
	9   MY_APP                                0x0000000106f39203 _ZN8facebook5reactL11invokeInnerEP9RCTBridgeP13RCTModuleDatajRKN5folly7dynamicE + 251
	10  MY_APP                                0x0000000106f38f85 ___ZN8facebook5react15RCTNativeModule6invokeEjON5folly7dynamicEi_block_invoke + 78
	11  libdispatch.dylib                   0x000000010e367595 _dispatch_call_block_and_release + 12
	12  libdispatch.dylib                   0x000000010e368602 _dispatch_client_callout + 8
	13  libdispatch.dylib                   0x000000010e36fb0b _dispatch_lane_serial_drain + 791
	14  libdispatch.dylib                   0x000000010e370784 _dispatch_lane_invoke + 428
	15  libdispatch.dylib                   0x000000010e37a89a _dispatch_workloop_worker_thread + 733
	16  libsystem_pthread.dylib             0x000000010e6fd611 _pthread_wqthread + 421
	17  libsystem_pthread.dylib             0x000000010e6fd3fd start_wqthread + 13
)
libc++abi.dylib: terminating with uncaught exception of type NSException
```
> react-native info
```
System:
    OS: macOS Mojave 10.14.6
    CPU: (4) x64 Intel(R) Core(TM) i5-5250U CPU @ 1.60GHz
    Memory: 317.59 MB / 8.00 GB
    Shell: 5.3 - /bin/zsh
  Binaries:
    Node: 13.13.0 - /usr/local/bin/node
    Yarn: Not Found
    npm: 6.14.4 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.7.5 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: iOS 12.1, macOS 10.14, tvOS 12.1, watchOS 5.1
    Android SDK:
      API Levels: 23, 24, 26, 28
      Build Tools: 28.0.3
      System Images: android-28 | Google Play Intel x86 Atom
      Android NDK: Not Found
  IDEs:
    Android Studio: Not Found
    Xcode: 10.1/10B61 - /usr/bin/xcodebuild
  Languages:
    Java: 1.8.0_73 - /usr/bin/javac
    Python: 2.7.10 - /usr/bin/python
  npmPackages:
    @react-native-community/cli: Not Found
    react: 16.11.0 => 16.11.0 
    react-native: 0.62.2 => 0.62.2 
  npmGlobalPackages:
    *react-native*: Not Found
```
Any help would be highly appreciated.
Thanks-in-advance.

**Comment by hesan-aminiloo** [comment]
I have the same issue. What I'm trying to do is upgrade from `0.60.4` to `0.62.2` and there are so many errors with Xcode.

![Screen Shot 2020-04-25 at 9 06 40 AM](https://user-images.githubusercontent.com/34710912/80271275-16af5700-86d4-11ea-8a02-027207d17f5f.png)

I'm using `react-native-firebase` version 5.x .
I guess some of these errors are related to the firebase package.

Any clue?

**Comment by cbothra** [comment]
After a lot of debugging process finally got the solution.

Actually in the project there was an npm package react-native-pie-chart which has a peer dependency of ART but since ART is now removed from react-native package it was failing. ART is now available as a standalone package '@react-native-community/art'.

So I removed the react-native-pie-chart package and added back using https://github.com/sarmad1995/react-native-pie-chart since the original react-native-pie-chart package is not updated yet and still using ART from react-native.
